File: Python/Stack.py
class Node:
    def __init__(self, value):
        self.data = value
        self.prev = None

class Stack:
    length = 0
    head = None

    def peek(self):
        return self.head.data

    def push(self, value):
        self.length += 1
        node = Node(value)
        if self.head == None:
            self.head = node
            return;
        node.prev = self.head
        self.head = node
    
    def pop(self):
        self.length = max(0, self.length - 1)
        if self.length == 0:
            tmp = self.head
            self.head = None
            tmp.prev = None
            return tmp.data
        tmp = self.head
        self.head = tmp.prev
        tmp.prev = None
        return tmp.data


stack = Stack()
stack.push(1)
print(stack.peek())
stack.push(2)
print(stack.peek())
stack.push(3)
print(stack.peek())
stack.push(4)
print(stack.peek())
stack.push(5)
print(stack.peek())
print(stack.pop())
print(stack.peek())
print(stack.pop())
print(stack.peek())
print(stack.pop())
print(stack.peek())
print(stack.pop())
print(stack.peek())
print(stack.pop())

File: Java/Stack.java
public class Stack {

    public static class Node{
        int data;
        Node prev;

        public Node(int value){
            data = value;
        }
    }

    public static class StackClass{
        private Node head;
        public int length = 0;

        public int peek(){
            return head.data;
        }

        public void push(int value){
            length++;
            Node node = new Node(value);
            if(head == null){
                head = node;
                return;
            }
            node.prev = head;
            head = node;
        }

        public int pop(){
            length = Math.max(0, length - 1);
            if(length == 0){
                Node tmp = head;
                head = null;
                tmp.prev = null;
                return tmp.data;
            }
            Node tmp = head;
            head = tmp.prev;
            tmp.prev = null;
            return tmp.data;
        }
    }

    public static void main(String[] args){
        StackClass stack = new StackClass();
        stack.push(1);
        System.out.println(stack.peek());
        stack.push(2);
        System.out.println(stack.peek());
        stack.push(3);
        System.out.println(stack.peek());
        stack.push(4);
        System.out.println(stack.peek());
        stack.push(5);
        System.out.println(stack.peek());
        System.out.println(stack.pop());
        System.out.println(stack.peek());
        System.out.println(stack.pop());
        System.out.println(stack.peek());
        System.out.println(stack.pop());
        System.out.println(stack.peek());
        System.out.println(stack.pop());
        System.out.println(stack.peek());
        System.out.println(stack.pop());
    }
}
